SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -409,365 +231,162 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
+	// Create the folder for storing the snapshots.
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	PlatformFile.CreateDirectoryTree(*SnapshotPath);
 
-			if (CallBack)
-			{
-				CallBack(bSuccess);
-			}
-		});
+	// Use the runtime start timestamp as the log directory, e.g. `<Project>/spatial/localdeployment/<timestamp>/`
+	FString LocalDeploymentLogsDir = FPaths::Combine(SpatialGDKServicesConstants::LocalDeploymentLogsDir, RuntimeStartTime.ToString());
 
-	return;
-}
+	// runtime.exe --config=squid_config.json --snapshot=snapshots/default.snapshot --worker-port 8018 --http-port 5006 --grpc-port 7777
+	// --worker-external-host 127.0.0.1 --snapshots-directory=spatial/snapshots/<timestamp>
+	// --schema-bundle=spatial/build/assembly/schema/schema.sb
+	// --event-tracing-logs-directory=`<Project>/spatial/localdeployment/<timestamp>/`
+	FString RuntimeArgs =
+		FString::Printf(TEXT("--config=\"%s\" --snapshot=\"%s\" --worker-port %s --http-port=%s --grpc-port=%s "
+							 "--snapshots-directory=\"%s\" --schema-bundle=\"%s\" --event-tracing-logs-directory=\"%s\" %s"),
+						*LaunchConfig, *SnapshotName, *FString::FromInt(WorkerPort), *FString::FromInt(HTTPPort),
+						*FString::FromInt(GRPCPort), *SnapshotPath, *SchemaBundle, *LocalDeploymentLogsDir, *LaunchArgs);
 
-bool FLocalDeploymentManager::TryStopLocalDeployment()
-{
-	if (!bLocalDeploymentRunning || LocalRunningDeploymentID.IsEmpty())
+	if (!RuntimeIPToExpose.IsEmpty())
 	{
-		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Tried to stop local deployment but no active deployment exists."));
-		return false;
+		RuntimeArgs.Append(FString::Printf(TEXT(" --worker-external-host %s"), *RuntimeIPToExpose));
 	}
 
-	bStoppingDeployment = true;
+	// Setup the runtime file logger.
+	SetupRuntimeFileLogger(LocalDeploymentLogsDir);
 
-	FString SpotDeleteArgs = FString::Printf(TEXT("alpha deployment delete --id=%s --json"), *LocalRunningDeploymentID);
+	FString RuntimePath = SpatialGDKServicesConstants::GetRuntimeExecutablePath(RuntimeVersion);
 
-	FString SpotDeleteResult;
-	FString StdErr;
-	int32 ExitCode;
-	FPlatformProcess::ExecProcess(*SpatialGDKServicesConstants::SpotExe, *SpotDeleteArgs, &ExitCode, &SpotDeleteResult, &StdErr);
-	bStoppingDeployment = false;
+	RuntimeProcess = { *RuntimePath, *RuntimeArgs, SpatialGDKServicesConstants::SpatialOSDirectory, /*InHidden*/ true,
+					   /*InCreatePipes*/ true };
 
-	if (ExitCode != ExitCodeSuccess)
-	{
-		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to stop local deployment! Result: %s - Error: %s"), *SpotDeleteResult,
-			   *StdErr);
-	}
+	FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+	TWeakPtr<SSpatialOutputLog> SpatialOutputLog = GDKServices.GetSpatialOutputLog();
 
-	bool bSuccess = false;
+	RuntimeProcess->OnOutput().BindLambda([&RuntimeLogFileHandle = RuntimeLogFileHandle, &bStartingDeployment = bStartingDeployment,
```

Comment:
Can the lambda capture be

&RuntimeLogFileHandle, &bStartingDeployment  

instead of 

&RuntimeLogFileHandle = RuntimeLogFileHandle, &bStartingDeployment = bStartingDeployment

Reply:
+1

Reply:
This was raised in the last PR and I responded. `RuntimeLogFileHandle` and `bStartingDeployment` are member variables and so cannot be captured in this way, you must make a copy of the address to the pointer in order to capture by reference (in C++ 14 and above). 
https://stackoverflow.com/questions/7895879/using-member-variable-in-lambda-capture-list-inside-a-member-function

The alternative is to capture `this` which is what I did originally but was asked to reduce what's captured :p